src/vehicle_gui.cpp

```diff
@@ -108,30 +171,63 @@ uint GetUnitNumberDigits(VehicleList &vehicles)
 		unitnumber = max<uint>(unitnumber, v->unitnumber);
 	}
 
-	if (unitnumber >= 10000) return 5;
-	if (unitnumber >=  1000) return 4;
-	if (unitnumber >=   100) return 3;
-
-	/*
-	 * When the smallest unit number is less than 10, it is
-	 * quite likely that it will expand to become more than
-	 * 10 quite soon.
-	 */
-	return 2;
+	return CountDigitsForAllocatingSpace(unitnumber);
 }
 
 void BaseVehicleListWindow::BuildVehicleList()
 {
-	if (!this->vehicles.NeedRebuild()) return;
+	if (!this->vehgroups.NeedRebuild()) return;
 
 	DEBUG(misc, 3, "Building vehicle list type %d for company %d given index %d", this->vli.type, this->vli.company, this->vli.index);
 
+	this->vehgroups.clear();
+
+	/* Note: This can be optimized so that merely changing the "group by" dropdown value does not need to call GenerateVehicleSortList again. */
```

Comment:
Not sure what I need to do with this comment ... should I expect you to pick this up? And I am pretty sure there are many many more places we can optimize things .. similar with some other comments, these comments only confuse us more in a year or in 2 years, than that they add value. I would suggest removing it.

Reply:
I guess... I put that there in case at some point we realise that GenerateVehicleSortList() runs too slowly... but will remove it.